server/client.go

```diff
@@ -390,6 +390,22 @@ func (c *client) initClient() {
 	}
 }
 
+// RemoteAddress expose the TCP Address of the client connection,
+// nil when not connected or unknown
+func (c *client) RemoteAddress() *net.TCPAddr {
+	tlsc, ok := c.nc.(*tls.Conn)
```

Comment:
If we chose to go with the `Addr` interface then it would be just returning `c.nc.RemoteAddr()`, but either way you need to:
. Do this under the `*client` object (`c.mu.Lock()/Unlock()`)
. Check that `c.nc` is not nil